### PR TITLE
new bot template

### PR DESCRIPTION
**Pull request by YitongFeng-git**
Animation

### CODE REVIEW

File: bot/js/bots/teamsBot.js [hunk 1]
```diff
@@ -19,7 +19,11 @@ class TeamsBot extends DialogBot {
             const membersAdded = context.activity.membersAdded;
             for (let cnt = 0; cnt < membersAdded.length; cnt++) {
                 if (membersAdded[cnt].id !== context.activity.recipient.id) {
-                    await context.sendActivity('Welcome to TeamsBot. Type \'login\' to get logged in. Type \'logout\' to sign-out.');
+                    const welcomeMsg = MessageFactory.text(`Congratulations! ${username}, your hello world Bot 
+                    template is running. This bot will introduce you how to build bot using Microsoft Teams App Framework(TeamsFx). 
+                    You can reply ‘intro’ to see the introduction card. TeamsFx helps you build Bot using [Bot Framework SDK](https://dev.botframework.com/)`);
```
Comment (henzhang-ms) [inline bot/js/bots/teamsBot.js:24]: why '‘' instead of '\''

Comment (henzhang-ms) [reply bot/js/bots/teamsBot.js:24]: and should be careful about the leading space and \n in the string

File: bot/js/dialogs/mainDialog.js [hunk 2]
```diff
@@ -48,9 +48,9 @@ class MainDialog extends LogoutDialog {
         if (results.status === DialogTurnStatus.empty) {
             await dialogContext.beginDialog(this.id);
         }
-    }
+    }
```
Comment (henzhang-ms) [inline bot/js/dialogs/mainDialog.js:51]: tailing space

File: bot/js/dialogs/rootDialog.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+            return result;
+        }
+
+        return await super.onContinueDialog(innerDc);
+    }
+
+    async interrupt(innerDc) {
+        const removedMentionText = TurnContext.removeRecipientMention(innerDc.context.activity, innerDc.context.activity.recipient.id);
+        if (removedMentionText) {
+            const text = removedMentionText.toLowerCase().replace(/\r?\n|\r/g, '');    // Remove the line break
```
Comment (henzhang-ms) [inline bot/js/dialogs/rootDialog.js:35]: why not just /[\r\n]/g

**Comment by YitongFeng-git** [comment]
### When first meet this Bot

image

File: bot/js/dialogs/mainDialog.js [hunk 4]
```diff
@@ -53,22 +55,13 @@ class MainDialog extends LogoutDialog {
...
     }
 
-    async promptStep(stepContext) {
-        try {
-            return await stepContext.beginDialog(TEAMS_SSO_PROMPT_ID);
-        } catch (err) {
-            console.error(err);
-        }
+    async ssoStep(stepContext) {
+        return await stepContext.beginDialog(TEAMS_SSO_PROMPT_ID);
```
Comment (IvanJobs) [inline bot/js/dialogs/mainDialog.js:59]: Nit: the reason to remove try/catch?

File: bot/js/dialogs/mainDialog.js [hunk 5]
```diff
@@ -79,32 +72,17 @@ class MainDialog extends LogoutDialog {
...
 
                 // show user picture
-                //var photoBuffer = await graphClient.api("/me/photo/$value").get();
-                // const photoBuffer =await photoResponse.arrayBuffer();
-                // const photoData = photoResponse.data;
-                // const imageUri = 'data:image/png;base64,' + photoData.toString('base64');
-                //const card = CardFactory.thumbnailCard("", CardFactory.images([imageUri]));
-                // await stepContext.context.sendActivity({ attachments: [card] });
+                var photoBinary = await graphClient.api("/me/photo/$value").responseType(ResponseType.ARRAYBUFFER).get();
+                //const photoBuffer =await photoResponse.arrayBuffer();
```
Comment (IvanJobs) [inline bot/js/dialogs/mainDialog.js:76]: Nit: maybe remove this line of comment?

File: bot/js/dialogs/mainDialog.js [hunk 6]
```diff
@@ -79,32 +72,17 @@ class MainDialog extends LogoutDialog {
...
+                const buffer = Buffer.from(photoBinary);
+                const imageUri = 'data:image/png;base64,' + buffer.toString('base64');
```
Comment (IvanJobs) [inline bot/js/dialogs/mainDialog.js:78]: Nit: using string interpolation would be better, I mean the anti-quote.

File: bot/js/dialogs/rootDialog.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                    'Introduction',
+                    null,
+                    cardButtons,
+                    {
+                        text: `This Bot has implemented single sign-on (SSO) using Teams Account 
+                            which user logged in Teams client, check <a href=\"placeholder\">TeamsFx authentication document</a> 
+                            and code in <pre>bot/dialogs/mainDialog.js</pre> to learn more about SSO.
+                            Type <strong>show</strong> or click the button below to show your profile by calling Microsoft Graph API with SSO.
+                            To learn more about building Bot using Microsoft Teams App Framework(TeamsFx), please refer to the <a href=\"placeholder\">TeamsFx document</a> .`
+                    });
```
Comment (IvanJobs) [inline bot/js/dialogs/rootDialog.js:49]: Nit: put this text into a file to improve readability.

**Review by IvanJobs — APPROVED** [review]
Some nit comments were added which may make the template better.

File: bot/js/dialogs/mainDialog.js (hunk 4 above)
Comment (YitongFeng-git) [reply bot/js/dialogs/mainDialog.js:59]: no need try catch, inner func has catched

File: bot/js/dialogs/mainDialog.js (hunk 5 above)
Comment (YitongFeng-git) [reply bot/js/dialogs/mainDialog.js:76]: ok

File: bot/js/dialogs/rootDialog.js (hunk 7 above)
Comment (YitongFeng-git) [reply bot/js/dialogs/rootDialog.js:49]: I think it is just a helloworld bot, put it here is more convenient for user to read.

File: bot/js/dialogs/mainDialog.js (hunk 6 above)
Comment (eriolchan) [reply bot/js/dialogs/mainDialog.js:75]: It looks that we use both single quote (') and double quote (") in our code, can we align them into one style?

**Pull request merged**